openedx/core/djangoapps/celery_utils/task.py

```diff
@@ -17,15 +17,20 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
         """
         If the task fails, persist a record of the task.
         """
-        FailedTask.objects.create(
-            task_name=_truncate_to_field(FailedTask, 'task_name', self.name),
-            task_id=task_id,  # Fixed length UUID: No need to truncate
-            args=args,
-            kwargs=kwargs,
-            exc=_truncate_to_field(FailedTask, 'exc', repr(exc)),
-        )
+        if not FailedTask.objects.filter(task_id=task_id, datetime_resolved=None).exists():
+            FailedTask.objects.create(
+                task_name=_truncate_to_field(FailedTask, 'task_name', self.name),
+                task_id=task_id,  # Fixed length UUID: No need to truncate
+                args=args,
+                kwargs=kwargs,
+                exc=_truncate_to_field(FailedTask, 'exc', repr(exc)),
+            )
         super(PersistOnFailureTask, self).on_failure(exc, task_id, args, kwargs, einfo)
 
+    def on_success(self, retval, task_id, args, kwargs):
+        FailedTask.objects.filter(task_id=task_id, datetime_resolved=None).update(datetime_resolved=now())
```

Comment:
Isn't this going to update the table even when there was no previous failure?  That may explain the unexpected increase in query count.